### PR TITLE
Add section on experimental cache to Content Collection page

### PR DESCRIPTION
**Pull request by natemoo-re**
#### Description (required)

Adds a new section to **Content Collection** page to highlight the forthcoming `experimental.contentCollectionCache` flag. This flag will automatically be added to the configuration reference, so that isn't covered here.

### CODE REVIEW

**Comment by natemoo-re** [comment]
I believe the lint check will fail until the config reference is updated!

File: src/content/docs/en/guides/content-collections.mdx [hunk 1]
```diff
@@ -542,6 +542,23 @@ If you have an existing Astro project, such as a blog, that uses Markdown or MDX
  
 See how to convert a basic blog example from `src/pages/posts/` to `src/content/posts` in our [step-by-step tutorial](/en/tutorials/add-content-collections/) that uses the codebase from [the Build a Blog tutorial's finished project](https://github.com/withastro/blog-tutorial-demo).
 
+## Enabling Build Caching
+
+<p>
+  <Since v="3.5.0" />
+</p>
+
+If you are working with large Content Collections, it is possible to enable cached builds with the new [`experimental.contentCollectionCache`](/en/reference/configuration-reference/#experimentalcontentcollectioncache) flag. This experimental feature optimizes Astro's build process, enabling unchanged Content Collections to be stored and reused between builds.
```
Comment (sarah11918) [inline src/content/docs/en/guides/content-collections.mdx:551]:
```suggestion
If you are working with large collections, you may wish to enable cached builds with the [`experimental.contentCollectionCache`](/en/reference/configuration-reference/#experimentalcontentcollectioncache) flag. This experimental feature optimizes Astro's build process, enabling unchanged collections to be stored and reused between builds.
```
nit: you can opt in even if you don't have large collections!  😄

**Review by sarah11918 — APPROVED** [review]
One small nit, but ready for tomorrow!  🥳

File: src/content/docs/en/guides/content-collections.mdx (hunk 1 above)
Comment (natemoo-re) [reply src/content/docs/en/guides/content-collections.mdx:551]: Great catch!

**Pull request merged**